### PR TITLE
Fixed segmentation fault from reinitting material faces 

### LINKED ISSUE #23348: Postprocessors and Materials Seg Fault
## Bug Description

The code below will segfault. I have a MWE below that will segfault if a material has `use_displaced_mesh = true` and the postprocessor does not. While the case below doesn't need the `use_displaced_mesh = true`, for the case I was working on the segfault sent me down a rabbit hole! ## Steps to Reproduce

```
[Mesh]
  [gen]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 16
    ny = 16
  []
[]

[GlobalParams]
  displacements = 'disp_x disp_y'
[]

[Modules/TensorMechanics/Master]
  [all]
    # decomposition_method = EigenSolution
    use_automatic_differentiation = false
    strain = FINITE
    add_variables = true
  []
[]
[Variables]
  [disp_x]
    order = FIRST
  []
  [disp_y]
    order = FIRST
  []
[]

[Materials]
  [elastic_tensor_cover]
    type = ComputeIsotropicElasticityTensor
    youngs_modulus = 10
    poissons_ratio = 0.3
    # if you comment out the code below then it will not segfault
    use_displaced_mesh = true
  []
  [stress]
    type = ComputeFiniteStrainElasticStress
  []
  [density]
    type = GenericConstantMaterial
    prop_names = 'density'
    prop_values = '1'
  []
[]


[Executioner]
  type = Transient
  dt = 0.3
  num_steps = 3
[]


[Postprocessors]
  [cause_of_failure]
    type = SideAverageValue
    boundary = right
    variable = disp_x
    # if you uncomment the line below, it will not segfault
    # use_displaced_mesh = true
  []
[]
```
## Impact

Probably should have some error to tell user that material and postprocessor settings are not compatible.

### PR DESCRIPTION
**Pull request by socratesgorilla**
Closes #23348

### CODE REVIEW

**Review by lindsayad — CHANGES_REQUESTED** [review]
It would be good to add the test input that @maxnezdyur provided to ensure we don't regress in the future

**Comment by socratesgorilla** [comment]
> It would be good to add the test input that @maxnezdyur provided to ensure we don't regress in the future

Just a RunApp or do you think it needs any actual output checking?

**Comment by lindsayad** [comment]
I think a `RunApp` is fine

File: modules/combined/test/tests/misc/re_init_face.i [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  []
+[]
+
+
+[Executioner]
+  type = Transient
+  dt = 0.3
+  num_steps = 3
+[]
+
```
Comment (lindsayad) [inline modules/combined/test/tests/misc/re_init_face.i:53]:
```suggestion
```

File: modules/combined/test/tests/misc/re_init_face.i [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+[Postprocessors]
+  [cause_of_failure]
```
Comment (lindsayad) [inline modules/combined/test/tests/misc/re_init_face.i:56]:
```suggestion
  [side_average]
```

File: modules/combined/test/tests/misc/re_init_face.i [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    type = SideAverageValue
+    boundary = right
+    variable = disp_x
+  []
+[]
+
```
Comment (lindsayad) [inline modules/combined/test/tests/misc/re_init_face.i:62]:
```suggestion
```

File: modules/combined/test/tests/misc/tests [hunk 4]
```diff
@@ -0,0 +1,9 @@
+[Tests]
+  design = 'Materials/index.md'
+  issues = '#23348'
+  [test_reinit_face]
+   type = RunApp
+   input = re_init_face.i
+   requirement = "The system shall properly reset itself after adding a material using a displaced mesh."
```
Comment (lindsayad) [inline modules/combined/test/tests/misc/tests:7]:
```suggestion
   requirement = "The system shall be able to perform requested material computation on faces of a displaced mesh even if no other face objects are executing on the displaced mesh."
```

**Review by lindsayad — CHANGES_REQUESTED** [review]

**Review by lindsayad — APPROVED** [review]

**Pull request merged**